src/QtBase/FileManager.cpp

```diff
@@ -179,8 +191,26 @@ void FileManager::OpenClassifier(ELocation location, const char* identifier, con
 		if (CloseWithPrompt(activeClassifier->GetUuid()) == false)
 			return;	// user aborted
 
+	if (mIsInFileSaving)
+	{
+		//wait for backend reply
+		QTimer timer;
+		timer.setSingleShot(true);
+		QEventLoop loop;
+		connect(this, &FileManager::writeToBackendFinished, &loop, &QEventLoop::quit);
+		connect(&timer, &QTimer::timeout, &loop, &QEventLoop::quit);
+		timer.start(10000);
+		loop.exec();
+		if (!timer.isActive())
+		{
+			Core::LogError("Timeout on writing data to backend");
```

Comment:
Have you actually tested this code branch?
Isn't some more resetting required here, e.g. at least `mIsInFileSaving = false` ?

Reply:
the code is tested, yes.  There is 'mIsInFileSaving = false;' at the end of the method. 
Do you think we may need to move it or add somewhere else?